src/main/ml-modules/root/com.marklogic.smart-mastering/impl/process.xqy

```diff
@@ -98,15 +98,25 @@ declare function proc-impl:consolidate-notifies($all-matches as map:map, $consol
           merge-impl:build-merge-uri(map:get($merged-into, $key), $key)
         else
           $key
-      for $key-notification in map:get($all-matches, $key)/result[@action="notify"]
-      let $key-uri as xs:string := $key-notification/@uri
-      let $updated-uri :=
-        if (map:contains($merged-into, $key-uri)) then
-          merge-impl:build-merge-uri(map:get($merged-into, $key-uri), $key-uri)
-        else
-          $key-uri
+      let $key-notifications := map:get($all-matches, $key)/result[@action=$const:NOTIFY-ACTION]
```

Comment:
Group all notifications for a set of documents into the same notification, instead of creating a notification for each pairing